Add delete folder flow

CrystalPea opened the pull request:
### Work done:

- Add a delete folder link to manage folder page
- Add delete_folder method to template folder api client
- Add delete folder GET/POST endpoint 
- Add Flash messages and error handling to the airport
- Tests

Pivotal ticket: https://www.pivotaltracker.com/story/show/161177455


### Screenshots:

#### Happy path:

<img width="995" alt="screen shot 2018-11-14 at 17 03 37" src="https://user-images.githubusercontent.com/20957548/48499044-717e2700-e82f-11e8-913a-878b9bbe273b.png">

<img width="995" alt="screen shot 2018-11-14 at 17 56 26" src="https://user-images.githubusercontent.com/20957548/48502154-c70a0200-e836-11e8-8e8d-941bcabc5c89.png">

<img width="999" alt="screen shot 2018-11-14 at 17 04 34" src="https://user-images.githubusercontent.com/20957548/48499048-717e2700-e82f-11e8-95d4-51b44e5f5d59.png">


#### Folder not empty:

<img width="994" alt="screen shot 2018-11-14 at 17 00 31" src="https://user-images.githubusercontent.com/20957548/48498786-ddac5b00-e82e-11e8-8950-6a4eb4591435.png">

servingUpAces commented on app/main/views/templates.py:
```diff
@@ -387,9 +391,45 @@ def manage_template_folder(service_id, template_folder_id):
+                url_for('.choose_template', service_id=service_id)
+            )
+        except HTTPError as e:
+            msg = "Folder is not empty"
```
We can check that the folder is empty before calling delete.

CrystalPea replied:
You're right 👍

servingUpAces commented on app/templates/views/templates/manage-template-folder.html:
```diff
@@ -25,7 +25,15 @@ <h1 class="heading-large">
+        service_id=current_service_id,
+        template_folder_id=template_folder_id
+      ),
+      delete_link_text="Delete this folder") }}
```
Do we want to only show the delete link if the folder is empty?

CrystalPea replied:
We could, and then some kind of message if it's not? Like "If you want to delete this folder, you need to first empty it"...?

servingUpAces replied:
We've decided it's better to show the button and let them get the `must be empty` message.

servingUpAces commented on app/main/views/templates.py:
```diff
@@ -387,6 +391,51 @@ def manage_template_folder(service_id, template_folder_id):
+        try:
+            template_folder_api_client.delete_template_folder(current_service.id, template_folder_id)
+
+            flash("'{}' folder has been removed".format(template_folder_name), 'info')
```
We don't need this message

karlchillmaid replied:
`Are you sure you want to delete the [folder name] folder?`
`Confirm`

– as discussed, no need for a confirmation message –

Just finalising the _can't delete_ message

karlchillmaid commented:
Can't delete message:

`You must empty this folder before you can delete it`
`Move or delete all templates and folders from the [folder name] folder and try again.`

CrystalPea commented:
So, I changed the content of some messages and got rid of final confirmation message. I updated the screenshots to reflect those changes.

In a separate pull request I will change delete template 'are you sure' message so it is consistent with delete folder message, as per consultation with Karl.

@karlchillmaid @yahoopete @servingUpAces are you happy with how it all looks now?

karlchillmaid commented:
@CrystalPea Content in the screenshots matches my previous comments and our discussion yesterday.

Good to know the _delete template_ flow will change to match this.

One change – please remove the quote marks around the folder name. We don't use quote marks around template names in the _delete template_ flow.

Not for now – perhaps we can revisit the option to add an extra description to the _can't delete_ message sometime?
